src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -647,22 +647,19 @@ public void SetMembrane(string membraneName)
         gui.UpdateMembraneButtons(Membrane.InternalName);
     }
 
-    public void SetRigidity(float rigidity)
+    public void SetRigidity(int rigidity)
     {
-        if (Math.Abs(Rigidity - rigidity) < MathUtils.EPSILON)
+        if (Math.Abs(Rigidity * 10 - rigidity) < MathUtils.EPSILON * 10)
             return;
 
-        var cost = (int)(Math.Abs(rigidity - Rigidity) / 2 * 100);
+        int cost = (int)(Math.Abs(rigidity - Rigidity * 10) / 2 * 10);
```

Comment:
Also calculating the cost between ints (and using a constant for MP per rigidity step) will be much nicer here.